### PR TITLE
Fix: ContentPreview metrics to include file fetching

### PR DESCRIPTION
**Pull request by DanDeMicco**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @DanDeMicco has signed the CLA. Thanks for the pull request!

**Comment by jholdstock** [comment]
Hello @jeremypress. I think you have tagged the wrong user here, I have nothing to do with this project :)

File: src/components/ContentPreview/ContentPreview.js [hunk 1]
```diff
@@ -480,6 +545,7 @@ class ContentPreview extends PureComponent<Props, State> {
      * @return {void}
      */
     fetchFileSuccessCallback = (file: BoxItem): void => {
+        this.fetchFileEndTime = performance.now();
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: IE11?

Comment (DanDeMicco) [reply src/components/ContentPreview/ContentPreview.js]: verified IE11 supports this 👍

File: src/components/ContentPreview/ContentPreview.js [hunk 2]
```diff
@@ -369,6 +386,47 @@ class ContentPreview extends PureComponent<Props, State> {
...
+
+        const totalFetchFileTime = Math.round(this.fetchFileEndTime - this.fetchFileStartTime);
+        const { rendering, conversion, preload } = previewTimeMetrics;
+
+        // We need to add in the total file fetch time to the rendering and total as preview
+        // does not do the files call. In the case the file is in the process of
+        // being converted, we need to add to conversion instead of the render
+        let totalConversion = conversion;
+        let totalRendering = rendering;
+        let totalPreload = preload;
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: can make totalPreload a constant and just to addition here...

Comment (priyajeet) [reply src/components/ContentPreview/ContentPreview.js]: I guess you can make all consts

```js
const hasConversion = isFinite(conversion);
const totalConversion = hasConversion ? conversion + totalFetchFileTime : conversion;
const totalRendering = !hasConversion && isFinite(rendering) ? rendering + totalFetchFileTime : rendering;
const totalRendering = isFinite(preload) ? preload + totalFetchFileTime : preload;
```

File: src/components/ContentPreview/ContentPreview.js [hunk 3]
```diff
@@ -69,7 +69,7 @@ type Props = {
     sharedLink?: string,
     sharedLinkPassword?: string,
     onError?: Function,
-    onMetric?: Function,
+    onMetric: Function,
```
Comment (priyajeet) [inline src/components/ContentPreview/ContentPreview.js]: wrappers/ContentPreview.js doesn't pass this btw

Comment (DanDeMicco) [reply src/components/ContentPreview/ContentPreview.js]: Yeah, it is added in ```defaultProp``` which defaults it to noop.

**Pull request merged**